templates/boxMenuItem.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import Adapt from 'core/js/adapt';
+import { classes, compile } from 'core/js/reactHelpers';
+
+export default function BoxMenuItem (props) {
+
+  const {
+    displayTitle,
+    _graphic,
+    body,
+    duration,
+    linkText,
+    _isVisited,
+    _isLocked,
+    _isComplete,
+    title,
+    _isOptional,
+    _nthChild,
+    _totalChild
+  } = props;
+
+  const _globals = Adapt.course.get('_globals');
+
+  const durationLabel = duration
+    ? _globals?._menu?._boxMenu?.durationLabel
+      ? `${_globals?._menu?._boxMenu?.durationLabel.trim()} ${duration}`
+      : duration
+    : '';
+
+  
+  const visited = _isVisited ? _globals?._accessibility?._ariaLabels?.visited : '';
+  const complete = _isComplete ? _globals?._accessibility?._ariaLabels?.complete : '';
+  const completion = complete || visited;
+  const locked = _isLocked ? _globals?._accessibility?._ariaLabels?.locked : linkText;
+  const optional = _isOptional ? _globals?._accessibility?._ariaLabels?.optional : '';
+  const itemCount = compile(_globals?._menu?._boxMenu?.itemCount || '', { _nthChild, _totalChild });
+  const ariaLabel = `${completion} ${locked} ${title}. ${itemCount}. ${optional}`;
```

Comment:
```suggestion
  const ariaLabel = [
    completion, locked, `${title}.`, `${itemCount}.`, ${optional}
  ].filter(Boolean).join(' ');
```
Only add spaces between non-null text.